src/sentry_scope.c

```diff
@@ -96,17 +96,26 @@ sentry__scope_unlock(void)
 }
 
 void
-sentry__scope_flush(const sentry_scope_t *scope)
+sentry__scope_flush_unlock(const sentry_scope_t *scope)
 {
+    bool did_unlock = false;
```

Comment:
hm, this is a bit fragile. Ideally, I would like to put the scope and the session onto the options, since that is reference counted, and all this is quite interconnected anyway. I don’t know why we had the scope separate in the first place.